packages/core/src/messagingProtocols/mls/E2EIdentityService/E2EIServiceInternal.ts

```diff
@@ -35,86 +34,50 @@ import {createNewOrder, finalizeOrder} from './Steps/Order';
     /** number of seconds the certificate should be valid */
     private readonly certificateTtl: number,
     private readonly keyPackagesAmount: number,
-    private readonly dispatchNewCrlDistributionPoints: (payload: NewCrlDistributionPointsPayload) => void,
-  ) {}
-
-  // ############ Public Functions ############
-
-  public static async getInstance(params?: InitParams): Promise<E2EIServiceInternal> {
-    if (!E2EIServiceInternal.instance) {
-      if (!params) {
-        throw new Error('E2EIServiceInternal is not initialized. Please call getInstance with params.');
-      }
-      const {
-        skipInit = false,
-        coreCryptClient,
-        apiClient,
-        e2eiServiceExternal,
-        keyPackagesAmount,
-        dispatchNewCrlDistributionPoints,
-        certificateTtl,
-      } = params;
-      E2EIServiceInternal.instance = new E2EIServiceInternal(
-        coreCryptClient,
-        apiClient,
-        e2eiServiceExternal,
-        certificateTtl,
-        keyPackagesAmount,
-        dispatchNewCrlDistributionPoints,
-      );
-      if (!skipInit) {
-        const {discoveryUrl, user, clientId} = params;
-        if (!discoveryUrl || !user || !clientId) {
-          throw new Error('discoveryUrl, user and clientId are required to initialize E2EIServiceInternal');
-        }
-        await E2EIServiceInternal.instance.init({clientId, discoveryUrl, user});
-      }
-    }
-    return E2EIServiceInternal.instance;
-  }
-
-  get acmeService(): AcmeService {
-    if (!this._acmeService) {
-      throw new Error('Error while trying to get AcmeService. E2EIServiceInternal has not been initialized');
-    }
-    return this._acmeService;
-  }
-
-  get initialData(): InitialData {
-    if (!this._initialData) {
-      throw new Error('Error while trying to get InitialData. E2EIServiceInternal has not been initialized');
-    }
-    return this._initialData;
+    private readonly initialData: InitialData,
+  ) {
+    const {discoveryUrl} = initialData;
+    this.acmeService = new AcmeService(discoveryUrl);
   }
 
   public async startCertificateProcess(hasActiveCertificate: boolean) {
-    // Step 0: Check if we have a handle in local storage
-    // If we don't have a handle, we need to start a new OAuth flow
     const identity = await this.initIdentity(hasActiveCertificate);
-    return this.startNewOAuthFlow(identity);
+
+    // Store the values in local storage for later use (e.g. in the continue flow)
+    const {orderUrl, authChallenges} = await this.getEnrollmentChallenges(identity);
+    const {
+      authorization: {keyauth, oidcChallenge},
+    } = authChallenges;
+
+    // store auth data for continuing the flow later on
+    const handle = await this.coreCryptoClient.e2eiEnrollmentStash(identity);
+    E2EIStorage.store.handle(Encoder.toBase64(handle).asString);
+    E2EIStorage.store.authData(authChallenges);
+    E2EIStorage.store.orderData({orderUrl});
```

Comment:
Storing data is now hoisted to the top level method of this class and grouped together